pkg/model/registry_test.go

```diff
@@ -0,0 +1,200 @@
+//go:build race
+
+package model
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sync"
+	"testing"
+)
+
+const testNamespace = "mynamespace"
+const otherNamespace = "othernamespace"
+
+var testCmA = v1.ObjectMeta{
+	Name:      "configmap-A",
+	Namespace: testNamespace,
+}
+
+var testCmB = v1.ObjectMeta{
+	Name:      "configmap-B",
+	Namespace: testNamespace,
+}
+
+// This fixture provides some coverage for the usage of namespace *and* name in identifying ObjectMeta entries.
+// Note that the name *intentionally* collides with testCmA, but the namespace is different.
+var testCmAOtherNamespace = v1.ObjectMeta{
+	Name:      "configmap-A",
+	Namespace: otherNamespace,
+}
+
+var testPvcA = v1.ObjectMeta{
+	Name:      "persistentvolumeclaim-A",
+	Namespace: testNamespace,
+}
+
+var testPvcB = v1.ObjectMeta{
+	Name:      "persistentvolumeclaim-B",
+	Namespace: testNamespace,
+}
+
+var testPvcC = v1.ObjectMeta{
+	Name:      "persistentvolumeclaim-C",
+	Namespace: testNamespace,
+}
+
+var testPvcD = v1.ObjectMeta{
+	Name:      "persistentvolumeclaim-D",
+	Namespace: testNamespace,
+}
+
+// NB: These tests mostly exist to exercise synchronization and detect regressions related to them via the
+// Golang race detector. See: https://go.dev/blog/race-detector
+// In short, add -race to the go test flags when running this.
+func Test_Registry_BasicOperations_ConcurrencyTest(t *testing.T) {
```

Comment:
@sunsingerus here's the minimal (not assertion-heavy) go test I mentioned, its purpose being to exercise the synchronization primitives and detect any regressions related to concurrency